roles/expirationd.lua

```diff
@@ -0,0 +1,302 @@
+    local conf = setmetatable(table.deepcopy(task_conf), {})
+    local params_map = {
+        space = {required = true, types = {"n", "s"}},
+        is_expired = {required = true, types = {"f"}},
+        is_master_only = {required = false, types = {"b"}},
+        options = {required = false, types = {"t"}},
+    }
+    for k, _ in pairs(conf) do
+        if type(k) ~= "string" then
+            error(role_name .. ": param must be a string")
+        end
+        if params_map[k] == nil then
+            error(role_name .. ": unsupported param " .. k)
+        end
+    end
+    local missed_functions = {}
+    for param, opts in pairs(params_map) do
+        if opts.required and conf[param] == nil then
+            error(role_name .. ": " .. param .. " is required")
+        end
+        if conf[param] ~= nil then
+            local res, ok, err = get_param(param, conf[param], opts.types)
+            if not ok then
+                 error(err)
+            end
+            if ok and res == nil and opts.types[1] == "f" then
+                table.insert(missed_functions, conf[param])
+            end
+            conf[param] = res
+        end
+    end
+
+    local missed_functions_opts
+    conf.options, missed_functions_opts = get_task_options(conf.options)
+    if missed_functions_opts ~= nil then
+        for _, func in pairs(missed_functions_opts) do
+            table.insert(missed_functions, func)
+        end
+    end
+    return conf, missed_functions
+end
+
+local function get_cfg(cfg)
+    local conf = setmetatable(table.deepcopy(cfg), {})
+    local params_map = {
+        metrics = {"b"},
+    }
+
+    for k, _ in pairs(conf) do
+        if type(k) ~= "string" then
+            error(role_name .. ": config option must be a string")
+        end
+        if params_map[k] == nil then
+            error(role_name .. ": unsupported config option " .. k)
+        end
+    end
+
+    for param, types in pairs(params_map) do
+        if conf[param] ~= nil then
+            local _, ok, err = get_param(param, conf[param], types)
+            if not ok then
+                error(err)
+            end
+        end
+    end
+
+    return conf
+end
+
+local function validate_config(conf_new)
+    local conf = conf_new or {}
+
+    for task_name, task_conf in pairs(conf) do
+        local _, ok, err = get_param("task name", task_name, {"s"})
+        if not ok then
+            error(err)
+        end
+        local _, ok, err = get_param("task params", task_conf, {"t"})
+        if not ok then
+            error(err)
+        end
+        local ok, ret = pcall(get_task_config, task_conf)
+        if not ok then
+            if task_name == "cfg" then
+                get_cfg(task_conf)
+            else
+                error(ret)
+            end
+        end
+    end
+
+    return true
+end
+
+local function load_task(task_conf, task_name)
+    local timeout = 1
+    fiber.name(role_name .. ": " .. task_name)
```

Comment:
Please, add more newlines between different blocks of code.

Reply:
changed